op-batcher/batcher/channel.go

```diff
@@ -109,41 +109,29 @@ func (s *channel) Timeout() uint64 {
 
 // updateInclusionBlocks finds the first & last confirmed tx and saves its inclusion numbers
 func (s *channel) updateInclusionBlocks() {
-	if len(s.confirmedTransactions) == 0 || !s.confirmedTxUpdated {
+	if len(s.confirmedTransactions) == 0 {
 		return
 	}
-	// If there are confirmed transactions, find the first + last confirmed block numbers
-	min := uint64(math.MaxUint64)
-	max := uint64(0)
+	s.minInclusionBlock = uint64(math.MaxUint64)
+	s.maxInclusionBlock = uint64(0)
 	for _, inclusionBlock := range s.confirmedTransactions {
-		if inclusionBlock.Number < min {
-			min = inclusionBlock.Number
-		}
-		if inclusionBlock.Number > max {
-			max = inclusionBlock.Number
-		}
+		s.minInclusionBlock = min(s.minInclusionBlock, inclusionBlock.Number)
+		s.maxInclusionBlock = max(s.maxInclusionBlock, inclusionBlock.Number)
 	}
-	s.minInclusionBlock = min
-	s.maxInclusionBlock = max
-	s.confirmedTxUpdated = false
 }
 
 // pendingChannelIsTimedOut returns true if submitted channel has timed out.
 // A channel has timed out if the difference in L1 Inclusion blocks between
 // the first & last included block is greater than or equal to the channel timeout.
 func (s *channel) isTimedOut() bool {
-	// Update min/max inclusion blocks for timeout check
-	s.updateInclusionBlocks()
 	// Prior to the granite hard fork activating, the use of the shorter ChannelTimeout here may cause the batcher
 	// to believe the channel timed out when it was valid. It would then resubmit the blocks needlessly.
 	// This wastes batcher funds but doesn't cause any problems for the chain progressing safe head.
 	return s.maxInclusionBlock-s.minInclusionBlock >= s.cfg.ChannelTimeout
 }
 
-// pendingChannelIsFullySubmitted returns true if the channel has been fully submitted.
+// isFullySubmitted returns true if the channel has been fully submitted.
 func (s *channel) isFullySubmitted() bool {
-	// Update min/max inclusion blocks for timeout check
-	s.updateInclusionBlocks()
```

Comment:
I'm not really sure why we needed to call this here anyway.

Reply:
While I am looking at this function and godoc: IIUC this is actually returning whether the channel is fully _confirmed_ not just fully submitted. Do you think it should be renamed?

Reply:
you're right, it's a no-op in the current implementation, because the call to `isTimedOut` happens first which updates the inclusion block variables

I believe the reason it was introduced originally in https://github.com/ethereum-optimism/optimism/commit/bc663a4ff7de1faaa8a102e908d93459a6182a54 was because the log on line 98 includes these variables so they wanted to ensure they were updated

re: `isFullyConfirmed` that would make sense, however we do have an external metric exposed called `fully_submitted` and i'm reluctant to change it here and make it inconsistent with the metric, and also reluctant to change the metric name

Reply:
Maybe a compromise is to keep the names but just update the godoc comment.

Reply:
added a small comment in parens, but let me know if you were thinking something different